proposals/dictionary-expressions.md

```diff
@@ -55,178 +55,86 @@ Choices here would have implications regarding potential syntactic ambiguities,
-
-```c#
-Dictionary<string, int> nameToAge = ["mads": 21, existingDict.MaxPair(), .. otherDict];
-
-// would be rewritten similar to:
-
-Dictionary<string, int> __result = new();
-__result["mads"] = 21;
-
-// Note: the below casts must be legal for the dictionary
-// expression to be legal
-var __t1 = existingDict.MaxPair();
-__result[(string)__t1.Key] = (int)__t1.Value;
-
-foreach (var __t2 in otherDict)
-    __result[(string)__t2.Key] = (int)__t2.Value;
-```
-
-Many rules for *dictionary expressions* will correspond to existing rules for *collection expressions*, just requiring aspects such as *element* and *iteration types* to be some `KeyValuePair<,>`.
-
-Note: Many rules in this spec will refer to types needing to be the same `KeyValuePair<,>` type.  This is an informal way of saying the types must have an identity conversion between them.  As such, `KeyValuePair<(int X, int Y), object>` would be considered the same type a `KeyValuePair<(int, int), object?>` for the purpose of these rules.
-
-With a broad interpretation of these rules, all of the following would be legal:
-
-```c#
-Dictionary<string, int> nameToAge1 = ["mads": 21, existingKvp]; // as would
-Dictionary<string, int> nameToAge2 = ["mads": 21, .. existingDict]; // as would
-Dictionary<string, int> nameToAge3 = ["mads": 21, .. existingListOfKVPS];
-```
-
-
-### Answered question 1
-
-Can a dictionary type value be created without using a key_value_pair_element?  For example are the following legal:
-
-```c#
-Dictionary<string, int> d1 = [existingKvp];
-Dictionary<string, int> d2 = [.. otherDict];
-```
+    This is just a simple augmentation on top of the existing collection expression rules.  In the above example, the code will be emitted as:
 
-Note: the element `KeyValuePair<K1,V1>` types need not be identical to the `KeyValuePair<K2,V2>` type of the destination dictionary type.  They simply must be convertible to the `V1 this[K1 key] { ... }` indexer provided by the dictionary.
+    ```c#
+    __result.Add(new KeyValuePair<string, int>("mads", 21));
+    ```
 
-Yes.  These are legal: [LDM-2024-03-11](https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-03-11.md#conclusions)
+2. Introduction of the *dictionary type*. *Dictionary types* are types that are similar to the existing *collection types*, with the additional requirements that they have an *element type* of some `KeyValuePair<TKey, TValue>` *and* have an indexer `TValue this[TKey] { ... }`. The former requirement ensures that `List<T>` is not considered a dictionary type, with its `int`-to-`T` indexer. The latter requirement ensures that `List<KeyValuePair<int, string>>` is not considered a dictionary type, with its `int`-to-`KeyValuePair<int, string>` indexer. `Dictionary<TKey, TValue>` passes both requirements.
```

Comment:
```suggestion
2. Introduction of the *dictionary type*. *Dictionary types* are types that are similar to the existing *collection types*, with the additional requirements that they have an *element type* of some `KeyValuePair<TKey, TValue>` *and* have an indexer `TValue this[TKey] { ... }`. The former requirement ensures that `List<T>` is not considered a dictionary type, as its element type is `T` not `KeyValuePair<,>`. The latter requirement ensures that `List<KeyValuePair<int, string>>` is also not considered a dictionary type, with its `int`-to-`KeyValuePair<int, string>` indexer (instead of an `int`-to-`string` indexer). `Dictionary<TKey, TValue>` passes both requirements.
```